src/DGtal/arithmetic/SternBrocot.ih

```diff
@@ -496,6 +496,139 @@ end() const
   static CFracSequence dummy;
   return ConstIterator( 0, dummy.end() );
 }
+
+template <typename TInteger, typename TQuotient>
+inline
+typename DGtal::SternBrocot<TInteger, TQuotient>::Fraction
+DGtal::SternBrocot<TInteger, TQuotient>::Fraction::
+median(const Fraction & g) const
+{
+  return Fraction(this->p()+g.p(),this->q()+g.q());
+}
+
+
+//----------------------------------------------------------------------------
+template <typename TInteger, typename TQuotient>
+inline
+typename DGtal::SternBrocot<TInteger, TQuotient>::Fraction
+DGtal::SternBrocot<TInteger, TQuotient>::Fraction::
+fractionOfSmallestDenominatorInBetween(const Fraction & other) const
+{
+  Fraction f(*this);  
+  Fraction g(other); 
+  
+  if(f>g)
+    {
+      Fraction tmp = f; f = g; g = tmp;
+    }
+  
+  Fraction res;
+  
+  int i = 0;
+  bool minf;
+  
+  Value uf, ug;
+  ConstIterator itf=f.begin(), itg=g.begin();
+  
+  if(abs(f.p()*g.q()-f.q()*g.p())==1)
+    return  f.median(g);
+  
+
+  // Compare each coefficient of the two fractions one by one.
+  for(itf = f.begin(), itg = g.begin(); itf != f.end() && itg != g.end() ; ++itf, ++itg)
+    {
+      uf = *itf;
+      ug = *itg;
+      if(uf.first == ug.first)
+	{
+	  res.push_back(std::make_pair(uf.first,i));
+	  i++;
+	}
+      else
+	{
+	  int coeff = (uf.first<ug.first)?uf.first:ug.first;
+	  if(i==f.k() && i == g.k())
+	    {
+	      if(abs(uf.first-ug.first) <= 1)
```

Comment:
NumberTraits<...>::ONE instead of 1 for exact computations ?
(same thing here with abs)